docker/production/Dockerfile

```diff
@@ -16,29 +16,71 @@ EXPOSE 9394
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash
-    # update the sources with the repos set up
+# use "--login" option, so that .bashrc is sourced in new shells
+SHELL ["/bin/bash", "--login", "-c"] 
+
+# Install Node.js
+# https://github.com/nodesource/distributions/issues/1583#issuecomment-1597489401
+# https://stackoverflow.com/a/57546198/
+# Unfortunately, we have to explicitly specify the node version here
+# and cannot use 16.x as we need to put the node binary into the PATH
+# and therefore require the exact version to find the folder
+ENV NODE_VERSION=16.20.1
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+ENV NVM_DIR=/root/.nvm
+RUN source "${NVM_DIR}/nvm.sh" && nvm install "${NODE_VERSION}" && \
+    nvm use "${NODE_VERSION}" && nvm alias default "${NODE_VERSION}"
+
+ENV NODE_PATH="${NVM_DIR}/versions/node/v${NODE_VERSION}/bin/"
+ENV PATH="${NODE_PATH}:${PATH}"
+
+RUN nvm current; node --version
+RUN npm --version
+
+# Install Yarn (see https://yarnpkg.com/getting-started/install)
+# https://github.com/nodejs/corepack#corepack-prepare--nameversion
+ENV YARN_VERSION=1.22.19
+RUN corepack enable
+RUN corepack prepare "yarn@${YARN_VERSION}" --activate
+RUN which yarn; yarn --version
+# even though this is not specified in the corepack documentation, we need to
+# run "set version", otherwise the app user will have an old version of
+# yarn avaialble instead of our specified version
+RUN yarn set version "${YARN_VERSION}"
+
+# Make Node.js tooling available for other users
+# see https://www.digitalocean.com/community/tutorials/how-to-install-node-js-with-nvm-node-version-manager-on-a-vps
+# this command copies the currently active nvm node tooling to /usr/local/
+# so that it can be used by the app user later
+RUN n=$(which node); n=${n%/bin/node}; chmod 755 $n/bin/node; chmod 755 $n/bin/yarn; cp -r $n/{bin,lib,share} /usr/local
+
+# Install other dependencies
 RUN apt-get update
-    # install all the dependencies
 RUN apt-get install -y --no-install-recommends \
         ffmpeg imagemagick pdftk ghostscript rsync shared-mime-info
-RUN apt-get install -y nodejs yarn
 
+# Setup ImageMagick
 RUN sed -i '/disable ghostscript format types/,+6d' /etc/ImageMagick-6/policy.xml
 
+# Manage users
 RUN groupadd -g 501 app && useradd -g 501 -u 501 -m -d /usr/src/app app && \
     mkdir /private /caches && chown app:app /private /caches
 
+
+# ============= Now switch to the app user
 WORKDIR /usr/src/app
 USER app
+
+# Log Node.js tooling versions
+RUN which node; node --version
+RUN which yarn; yarn --version
+
 COPY --from=build-pdfcomprezzor /go/src/pdfcomprezzor.wasm /go/src/wasm_exec.js /usr/src/app/public/pdfcomprezzor/
 
 COPY --chown=app:app ./Gemfile ./Gemfile.lock ./yarn.lock ./package.json /usr/src/app/
+RUN bundle install
+RUN yarn install
```

Comment:
I dont know if this is actually the same as `yarn install --production=true` as it seems to take  this from the env then:
https://classic.yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production-true-false